Upstream: mail sent to the bug reporting addresses on
          http://links.twibright.com/development.php

From 8ede2e88dbfa841523cd0ec26228c5927dd618ff Mon Sep 17 00:00:00 2001
From: Johannes Nixdorf <mixi@exherbo.org>
Date: Sat, 13 Jun 2015 15:43:54 +0200
Subject: [PATCH 1/3] Support libressl by making the usage of RAND_edg
 conditional

---
 acconfig.h   | 3 +++
 configure.in | 7 +++++++
 https.c      | 5 +++++
 3 files changed, 15 insertions(+)

diff --git a/acconfig.h b/acconfig.h
index 6b23847..477a369 100644
--- a/acconfig.h
+++ b/acconfig.h
@@ -145,6 +145,9 @@
 /* */
 #undef HAVE_OPENSSL
 
+/* Define if the libcrypto has RAND_egd */
+#undef HAVE_RAND_EGD
+
 /* */
 #undef HAVE_NSS
 
diff --git a/configure.in b/configure.in
index 07d98e6..313a6d3 100644
--- a/configure.in
+++ b/configure.in
@@ -716,6 +716,13 @@ if test -z "$disable_ssl"; then
 		AC_MSG_RESULT($cf_result)
 	fi
 
+	if test "$cf_result" != no; then
+		AC_MSG_CHECKING([whether OpenSSL supports RAND_egd])
+		AC_TRY_LINK([#include <openssl/rand.h>], [RAND_egd(0)], have_rand_egd=yes, have_rand_egd=no)
+		test "$have_rand_egd" = yes && AC_DEFINE(HAVE_RAND_EGD)
+		AC_MSG_RESULT($have_rand_egd)
+	fi
+
 	if test "$cf_result" = no && test "$ssld" = yes -o "$ssld" = nss; then
 		if test "$cf_openssl_pkgconfig" = no; then
 			pkgconfig_nss=no
diff --git a/https.c b/https.c
index 9a38804..fcf9d5c 100644
--- a/https.c
+++ b/https.c
@@ -25,6 +25,11 @@
 
 #ifdef HAVE_SSL
 
+/* LibreSSL doesn't have RAND_egd */
+#ifndef HAVE_RAND_EGD
+#define RAND_egd(x) -1
+#endif
+
 static SSL_CTX *context = NULL;
 
 SSL *getSSL(void)
-- 
2.4.3

